src/interfaces/uproxy_core_api.ts

```diff
@@ -215,14 +217,29 @@ export interface InvitationData {
 
 export enum PortControlSupport {PENDING, TRUE, FALSE};
 
-// Argument to #cloudInstall.
-export interface CloudInstallArgs {
-  // Use this cloud computing provider to create a server.
-  providerName: string;
-  // Provider-specific region in which to locate the new server.
-  region: string;
+export enum CloudOperationType {
+  CLOUD_INSTALL = 0,
+  CLOUD_DESTROY = 1
+}
+
+// Arguments to cloudUpdate
+export interface CloudOperationArgs {
+  operation: CloudOperationType;
+  // Use this cloud computing provider to access a server.
+  providerName :string;
+  // Provider-specific region in which to locate a new server.
+  region ?:string;
```

Comment:
@trevj - is there some other way we can make union types in TypeScript?  I was hoping there could some way that if the operation was CLOUD_INSTALL then region would be required, but if operation was CLOUD_INSTALL region would not be used